File: day10/1_Reverse_Vowels.java
/*
Auther - Saurabh Verma
Problem link - https://leetcode.com/problems/reverse-vowels-of-a-string/
*/
class Solution {
    public String reverseVowels(String s) {
        char[] ch = s.toCharArray();
        int left = 0;
        int right = s.length() - 1;
        HashSet<Character> set = new HashSet<>(Arrays.asList('a','e','i','o','u','A','E','I','O','U'));
        while(left< right){
            if(set.contains(ch[left]) && set.contains(ch[right])){
                char c = ch[left];
                ch[left] = ch[right];
                ch[right] = c;
                left++; right--;
            }else if(set.contains(ch[left])){
                right--;
            }else if(set.contains(ch[right])){
                left++;
            }else{
                left++;
                right--;
            }
        }
        return new String(ch);
    }
}

File: day10/1_reverse_vowels.py
# created by KUMAR SHANU

# 1. Reverse Vowels of a String
# https://leetcode.com/problems/reverse-vowels-of-a-string/


class Solution:
    def reverseVowels(self, s: str) -> str:
        arr = list(s)
        l, r = 0, len(arr) - 1
        vowels = ('a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U')
        if s == ' ' * len(s):
            return s
        while l <= r:
            if arr[l] not in vowels:
                l += 1
                continue
            if arr[r] not in vowels:
                r -= 1
                continue

            arr[l], arr[r] = arr[r], arr[l]
            l += 1
            r -= 1

        return ''.join(arr)
